### PR TITLE
fix deprecated warning during check

### LINKED ISSUE #141: GitHub Authentication API deprecation
In the final check.rb script, after giving my personal GitHub password, I get 2 e-mails saying this GitHub API feature will be deprecated this year.

```

you used a password to access an endpoint through the GitHub API using Octokit Ruby Gem 4.18.0:

https://api.github.com/user

Basic authentication using a password to the API is deprecated and will soon no longer work. Visit https://developer.github.com/changes/2020-02-14-deprecating-password-auth/ for more information around suggested workarounds and removal dates.

Thanks,
The GitHub Team

----------------

On April 7th, 2020 at 18:27 (UTC) you or an application you used recently accessed the deprecated Authorizations endpoint on the GitHub API with the useragent Octokit Ruby Gem 4.18.0.

We will remove the Authorizations API endpoint on November 13, 2020. If you accessed the API via password authentication, then we recommend you use the web flow to authenticate. Please check that your app uses the web flow for authentication https://developer.github.com/apps/building-oauth-apps/authorizing-oauth-apps/#web-application-flow

You can learn more about these changes by visiting our developer blog https://developer.github.com/changes/2020-02-14-deprecating-oauth-auth-endpoint/

Thanks,
The GitHub Team
```

This can be updated as suggested in the link: https://developer.github.com/changes/2020-02-14-deprecating-password-auth/

### PR DESCRIPTION
**Pull request by barangerbenjamin**
Fix #141 
According to https://github.com/octokit/octokit.rb#two-factor-authentication

### CODE REVIEW

File: check.rb [hunk 1]
```diff
@@ -67,7 +68,8 @@ def check_all
         client.authorizations.find { |a| a[:note] == GITHUB_AUTHORIZATION_NOTE } ||
         client.create_authorization(
           scopes: ['user:email'],
-          note: GITHUB_AUTHORIZATION_NOTE)
+          note: GITHUB_AUTHORIZATION_NOTE,
+          headers: {'X-Github-OTP': GITHUB_AUTHORIZATION_OTP})
```
Comment (ssaunier) [inline check.rb:72]: Thanks for taking a stab at it! Did you try it enabling 2-factor on your own GitHub account? I can't see how thsi can work, for me you need to send a value coming from the OTP generator in the `X-GitHub-OTP` header! That means having a third `puts "If you have 2-factor enabled, please type the 6-digits code, or just press <Enter> if you don't"` and then having a `gets.chomp` reading that. That way, if it's `""`, you don't send the `X-Github-OTP` header, if it has a value then you do send it!

Comment (barangerbenjamin) [reply check.rb:72]: image

Now just need to test if it's still works without 2FA 😅

Comment (barangerbenjamin) [reply check.rb]: Works without too 🙌 
image

**Comment by ssaunier** [comment]
:+1:

**Pull request merged**